File: deepsearch_agents/deepsearch_web.py
"""DeepSearch 网站定向检索代理。"""

from __future__ import annotations

from typing import Any, Dict, Optional

from deepsearch_mcp.client import DeepSearchMCPClient, SearchResult
from source.api import DeepSearchTransport


class DeepSearchWebAgent:
    """封装站点定向检索逻辑。"""

    def __init__(
        self,
        *,
        client: Optional[DeepSearchMCPClient] = None,
        transport: Optional[DeepSearchTransport] = None,
    ) -> None:
        self._client = client
        self._transport = transport
        if self._client is None:
            if self._transport is None:
                self._transport = DeepSearchTransport.from_env()
            self._client = DeepSearchMCPClient(self._transport, tool_name="deepsearch-web")

    def search(
        self,
        query: str,
        *,
        top_k: int = 5,
        locale: str = "zh-CN",
        filters: Optional[Dict[str, Any]] = None,
    ) -> SearchResult:
        filters = filters or {}
        if not (filters.get("site") or filters.get("time_range")):
            raise ValueError("deepsearch-web 需要提供 site 或 time_range 过滤条件")
        return self._client.search(
            query,
            top_k=top_k,
            locale=locale,
            filters=filters,
        )

    def close(self) -> None:
        if self._transport is not None:
            close = getattr(self._transport, "close", None)
            if callable(close):
                close()

    def __enter__(self) -> "DeepSearchWebAgent":  # pragma: no cover - 便利接口
        return self

    def __exit__(self, *exc_info: Any) -> None:  # pragma: no cover - 便利接口
        self.close()

    @property
    def client(self) -> DeepSearchMCPClient:
        return self._client

File: deepsearch_agents/deepsearch_web.ts
import { DeepSearchMCPClient, type SearchOptions, type SearchResult } from "../deepsearch_mcp/client.js";
import { DeepSearchTransport } from "../source/api.js";
import { logger, type Logger } from "../source/logger.js";

interface AgentOptions {
  client?: DeepSearchMCPClient;
  transport?: DeepSearchTransport;
}

export class DeepSearchWebAgent {
  private readonly client: DeepSearchMCPClient;
  private readonly transport?: DeepSearchTransport;
  private readonly logger: Logger;

  constructor(options: AgentOptions = {}) {
    const agentLogger = logger.child({ agent: "deepsearch-web" });
    agentLogger.debug("初始化 DeepSearchWebAgent", { providedClient: Boolean(options.client) });

    if (options.client) {
      this.client = options.client;
      this.transport = options.transport;
    } else {
      const transport = options.transport ?? DeepSearchTransport.fromEnv();
      this.client = new DeepSearchMCPClient(transport, { toolName: "deepsearch-web" });
      this.transport = transport;
    }
    this.logger = agentLogger;
  }

  async search(query: string, options: SearchOptions = {}): Promise<SearchResult> {
    const filters = options.filters ?? {};
    if (!filters.site && !filters.time_range) {
      this.logger.error("缺少必要过滤条件", { query, options });
      throw new Error("deepsearch-web 需要提供 site 或 time_range 过滤条件");
    }

    this.logger.info("执行定向检索", { query, options });
    return this.client.search(query, { ...options, filters });
  }

  close(): void {
    this.logger.debug("关闭代理");
    this.transport?.close();
  }
}
